ppa/archive/models.py

```diff
@@ -121,6 +122,76 @@ def stats():
         return stats
 
 
+class ProtectedFlags(Flags):
+    '''Flag set of fields that should be protected if edited in the admin.'''
+    title = ()
+    subtitle = ()
+    sort_title = ()
+    enumcron = ()
+    author = ()
+    pub_place = ()
+    publisher = ()
+    pub_date = ()
+
+    @classmethod
+    def deconstruct(cls):
+        '''Give Django information needed to make
+        :class:`ProtectedField.no_flags` default in migration.'''
+        # (import path, [args], kwargs)
+        return ('ppa.archive.models.ProtectedFlags', ['no_flags'], {})
+
+    @classmethod
+    def all_fields(cls):
+        return list(ProtectedFlags.__members__.keys())
+
+    def as_list(self):
+        '''Return combined flags as a list of field names.'''
+        all_fields = list(self.__members__.keys())
+        return [field for field in all_fields
+                if getattr(self, field)]
+
+    def __str__(self):
+        protected_fields = self.as_list()
+        verbose_names = []
+        for field in protected_fields:
+            field = DigitizedWork._meta.get_field(field)
+            # if the field has no actual verbose name, just capitalize
```

Comment:
There isn't an existing method we can use here?

Reply:
Sadly, no. The `_meta` is how you get access to field names and they are pretty low-level. Thankfully no no longer an issue.